src/util/address.rs

```diff
@@ -191,8 +192,8 @@ impl Address {
         Address {
             network,
             addr_type: AddressType::Integrated(payment_id),
-            public_spend,
-            public_view,
+            public_spend: public_spend.compress(),
```

Comment:
:question: As a general question, why should we store compressed points? I would assume that one could wait until serialisation.

Reply:
Unfortunately, `EdwardsPoint`s don't implement `Hash` so we would have to find a solution around that. Compressing is not cheap but I guess doing it inside `Hash` might be fine?

Reply:
True, you have to compress before you hash. 

On principle, I would delay compressing until it's strictly necessary. The thing about `CompressedEdwardsY` is that it's no different to 32 bytes, so it can hold data that doesn't actually map to and `EdwardsPoint` (see `decompress`).

Reply:
Yeah I think moving the compressing to a manual implementation of `Hash` is probably the better idea.